gnucash/import-export/qif-imp/qif-file.scm

```diff
@@ -525,18 +507,33 @@
       private-retval))
 
 
-  (gnc:backtrace-if-exception
+  (catch #t
     (lambda ()
-      (let ((retval #f))
+      (let ((retval #f)
+            (conv (port-conversion-strategy #f)))
+
+        (set-port-conversion-strategy! #f 'error)
         ;; Safely read the file.
-        (set! retval (gnc:backtrace-if-exception private-read))
+        (set! retval (private-read))
+        (set-port-conversion-strategy! #f conv)
 
         ;; Fill the progress dialog.
-        (if (and progress-dialog
-                 (list? retval))
+        (when (and progress-dialog (list? retval))
           (gnc-progress-dialog-set-value progress-dialog 1))
 
-        retval))))
+        retval))
+
+    (lambda (key . rest)
+      (let ((msg (string-append
+                  (case key
+                    ((decoding-error)
+                     (format #f (_ "The file contains values that are not \
```

Comment:
s/values/characters :)

Reply:
A bit odd leaving review comments on your own PR! ;-)

Reply:
I reviewed myself at work and forgot to update!